### PR TITLE
Remove wrapErrorAndLog and usages for ThrottlingWarning and SummarizerWarning

### PR DESCRIPTION
**Pull request by markfields**
wrapErrorAndLog offers a generic solution where it's not really what anyone will want.  If someone wants to wrap an error unilaterally (which is very rare), they should also own figuring out what they need to log between the inner and outer error.

This has the biggest impact on `Summarizer` error handling/logging.  I replaced the `WrapError` log with a new `SummarizerRunFailed` event, not entirely sure it will be useful.  @pleath if you have an hour next week I'd love to walk through the code and Kusto and think a bit about how errors are handled and logged in the Summarizer, it's a bit hard to follow today and if there's any tweaks to what I do instead here we can make those changes now.

As for ThrottlingWarning, all the inner errors here will be logged elsewhere now - see event `DeltaConnectionFailureToConnect`

### CODE REVIEW

**Comment by markfields** [comment]
Oops need to merge main, closing and will reopen.

**Review by markfields — COMMENTED** [review]
@heliocliu - Can you take a look at the api.md files and my comments and help me nail down what needs to be reverted and done in next branch?  Thanks!

File: api-report/container-runtime.api.md [hunk 1]
```diff
@@ -511,8 +511,6 @@ export interface ISummarizerRuntime extends IConnectableRuntime {
 export interface ISummarizingWarning extends ContainerWarning {
     // (undocumented)
     readonly errorType: "summarizingError";
-    // (undocumented)
-    readonly logged: boolean;
```
[the remark this answers is not in the document]
Comment (heliocliu) [reply api-report/container-runtime.api.md:515]: Is `ISummarizingWarning` intended to be used exclusively internally?  If so you only need to worry about the runtime compatibility (so probably would still need to make this optional).  Otherwise yes this is a major breaking change, but it would be fine if you were to make it optional.

File: api-report/telemetry-utils.api.md [hunk 2]
```diff
@@ -309,9 +309,6 @@ export class ThresholdCounter {
 // @public
 export function wrapError<T extends LoggingError>(innerError: unknown, newErrorFn: (message: string) => T): T;
 
-// @public
-export function wrapErrorAndLog<T extends LoggingError>(innerError: unknown, newErrorFn: (message: string) => T, logger: ITelemetryLogger): T;
```
[the remark this answers is not in the document]
Comment (heliocliu) [reply api-report/telemetry-utils.api.md:313]: adding a new standalone function is fine in main from an api perspective

[the remark this answers is not in the document]
Comment (markfields) [reply api-report/telemetry-utils.api.md:313]: What about removing and existing one?

[the remark this answers is not in the document]
Comment (heliocliu) [reply api-report/telemetry-utils.api.md:313]: wtf i'm so blind that's red not green
if it was only used internally within fluid it's fine, otherwise it's breaking (but removing anything is breaking by default generally)

File: packages/loader/container-utils/src/error.ts [hunk 3]
```diff
@@ -63,11 +62,10 @@ export class ThrottlingWarning extends LoggingError implements IThrottlingWarnin
     static wrap(
         error: unknown,
         retryAfterSeconds: number,
-        logger: ITelemetryLogger,
     ): IThrottlingWarning {
         const newErrorFn =
             (errMsg: string) => new ThrottlingWarning(errMsg, retryAfterSeconds);
-        return wrapErrorAndLog(error, newErrorFn, logger);
+        return wrapError(error, newErrorFn);
```
Comment (tyler-cai-microsoft) [inline packages/loader/container-utils/src/error.ts:68]: Out of curiosity, what does wrap error do and what does wrap error and log do? Is it just a difference of logging?

**Comment by markfields** [comment]
Closing for now.  Worth revisiting but not urgent.  Might open a bug/task for it.
